src/hurry/workflow/tests/test_edgecases.py

```diff
@@ -0,0 +1,146 @@
+import unittest
+from hurry.workflow import interfaces, workflow
+
+from zope import component
+from zope.annotation.interfaces import IAttributeAnnotatable
+from zope.annotation import attribute
+from zope.annotation import interfaces as annotation_interfaces
+
+from zope.interface import implementer, Attribute
+
+from hurry.workflow.tests.test_doctest import WorkflowVersions
+
+
+"""Increase test coverage without polluting doctest."""
+
+
+class IDocument(IAttributeAnnotatable):
+    title = Attribute('Title')
+
+
+@implementer(IDocument)
+class Document(object):
+
+    def __init__(self, title):
+        self.title = title
+
+
+class WorkflowTestCase(unittest.TestCase):
+
+    def test_getTransitions_empty(self):
+        wf = workflow.Workflow([])
+        self.assertEqual([], wf.getTransitions(None))
+
+
+class WorkflowInfoTestCase(unittest.TestCase):
+
+    def setUp(self):
+        component.provideAdapter(
+            workflow.WorkflowState,
+            (annotation_interfaces.IAnnotatable,),
+            interfaces.IWorkflowState)
+        component.provideAdapter(
+            workflow.WorkflowInfo,
+            (annotation_interfaces.IAnnotatable,),
+            interfaces.IWorkflowInfo)
+        component.provideAdapter(
+            attribute.AttributeAnnotations,
+            (annotation_interfaces.IAttributeAnnotatable,),
+            annotation_interfaces.IAnnotations)
+        component.provideUtility(
+            WorkflowVersions(),
+            interfaces.IWorkflowVersions)
```

Comment:
This setup is polluting global state.  Can you please add a tearDown?

Using `zope.component.testing.PlacelessSetup` as a base class instead of `unittest.TestCase` would suffice.